#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.10958663387687151 0.10336352155642804 -0.988588262163268 1.6406542264857054
  position 0.49726196536824896 5.939310558892279 1.8683747415262364
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
Robot {
  translation 0 0 0.118
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 0.5
        metalness 0.2
      }
      geometry Mesh {
        url [
          "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chassis.dae"
        ]
      }
    }
    Solid {
      translation 0 0 0.39062
      children [
        Shape {
          appearance DEF DARKGREY PBRAppearance {
            baseColor 0.5 0.5 0.5
            metalness 0.9
          }
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chapas_inox_tapas.stl"
            ]
          }
        }
      ]
      name "base_chapa"
    }
    Solid {
      translation 0.359 0 0.30062
      children [
        Shape {
          appearance USE DARKGREY
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/summit_xls_chapas_inox_tapas_front.stl"
            ]
          }
        }
      ]
      name "base_chapa_front"
    }
    Solid {
      translation 0 0.28931 0.29062
      children [
        DEF LOGO Shape {
          appearance USE DARKGREY
          geometry Mesh {
            url [
              "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/robotnik_logo_chasis.stl"
            ]
          }
        }
      ]
      name "base_logo_left"
    }
    Solid {
      translation 0 -0.28931 0.29062
      rotation 0 0 1 3.141593
      children [
        USE LOGO
      ]
      name "base_logo_right"
    }
    Solid {
      translation 0.345 0 0
      rotation 0 0 -1 1.570796
      children [
        USE LOGO
      ]
      name "base_logo_front"
    }
    Solid {
      translation -0.345 0 0
      rotation 0 0 1 1.570796
      children [
        USE LOGO
      ]
      name "base_logo_rear"
    }
    Solid {
      translation 0.31 0 0.411
      scale 0.5 0.5 0.5
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "lidar"
    }
    Lidar {
      translation 0.31 0 0.411
      name "lidar(1)"
      numberOfLayers 1
      minRange 0.1
      maxRange 2
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.2225 -0.2045 0
      }
      device [
        RotationalMotor {
          name "front_right_wheel_joint"
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "front_right_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.2224991262236524 -0.20449996432578646 0.003676350005529974
        rotation 1 -8.981784796070379e-12 9.016178626542165e-12 1.5708000003496634
        children [
          DEF INTERIOR_WHEEL Group {
            children [
              Shape {
                appearance DEF WHEEL_APPEARANCE PBRAppearance {
                  baseColor 0.381347 0.381347 0.381347
                  metalness 0.4
                  emissiveColor 0.202 0.202 0.202
                }
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rims_int.dae"
                  ]
                }
              }
              Shape {
                appearance DEF ROLLERS_APPEARANCE PBRAppearance {
                  baseColor 0.08132 0.08132 0.08132
                  roughness 0.8
                  metalness 0
                }
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rollers_int.dae"
                  ]
                }
              }
            ]
          }
        ]
        name "front_right_wheel_link"
        contactMaterial "InteriorWheelMat"
        boundingObject DEF WHEEL_BO Cylinder {
          height 0.08
          radius 0.123
        }
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.2225 0.2045 0
      }
      device [
        RotationalMotor {
          name "front_left_wheel_joint"
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "front_left_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation 0.22249912660342347 0.2045000356580684 0.0036746173195968262
        rotation 1 -1.013287732384401e-11 1.0167271437138332e-11 1.5708000003496396
        children [
          DEF EXTERIOR_WHEEL Group {
            children [
              Shape {
                appearance USE WHEEL_APPEARANCE
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rims_ext.dae"
                  ]
                }
              }
              Shape {
                appearance USE ROLLERS_APPEARANCE
                geometry Mesh {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/omni_wheel_rollers_ext.dae"
                  ]
                }
              }
            ]
          }
        ]
        name "front_left_wheel_link"
        contactMaterial "ExteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2225 0.2045 0
      }
      device [
        RotationalMotor {
          name "back_left_wheel_joint"
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "back_left_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.22250086290585847 0.20450003522517907 0.003628388348279249
        rotation 1 2.805165095639736e-11 -2.801748550616189e-11 1.5708000003491327
        children [
          USE INTERIOR_WHEEL
        ]
        name "back_left_wheel_link"
        contactMaterial "InteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2225 -0.2045 0
      }
      device [
        RotationalMotor {
          name "back_right_wheel_joint"
          maxVelocity 100
          maxTorque 100
        }
        PositionSensor {
          name "back_right_wheel_joint_sensor"
        }
      ]
      endPoint Solid {
        translation -0.22250086332063382 -0.20449996475867724 0.0036301210342035567
        rotation 1 3.338081391661134e-11 -3.3346664547248625e-11 1.5708000003491567
        children [
          USE EXTERIOR_WHEEL
        ]
        name "back_right_wheel_link"
        contactMaterial "ExteriorWheelMat"
        boundingObject USE WHEEL_BO
        physics Physics {
          density -1
          mass 6.5
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            0.0303095 0.05241925 0.0303095
            0 0 0
          ]
        }
      }
    }
    Solid {
      translation 0.35 0 0.09
      name "base_docking_contact"
    }
    DEF IMU Transform {
      translation 0.127 -0.129 0.212
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.001 0.001 0.001
          }
        }
        Accelerometer {
        }
        Gyro {
        }
        Compass {
        }
      ]
    }
    DEF RGBD Solid {
      translation 0.3585 0 0.29062
      children [
        Solid {
          translation 0.007 0 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.3 0.3 0.3
                roughness 1
                roughnessMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/appearances/protos/textures/plastic/plastic_rough_roughness.jpg"
                  ]
                }
                metalness 0
                normalMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/appearances/protos/textures/plastic/plastic_rough_normal.jpg"
                  ]
                }
              }
              geometry Mesh {
                url [
                  "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/orbbec_astra.dae"
                ]
              }
            }
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                roughness 1
                roughnessMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/appearances/protos/textures/plastic/plastic_rough_roughness.jpg"
                  ]
                }
                metalness 0
                normalMap ImageTexture {
                  url [
                    "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/appearances/protos/textures/plastic/plastic_rough_normal.jpg"
                  ]
                }
              }
              geometry Mesh {
                url [
                  "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/robots/robotnik/summit_xl_steel/protos/meshes/orbbec_astra_cover.dae"
                ]
              }
            }
          ]
          name "front_rgbd_camera_link"
          boundingObject Transform {
            translation -0.02 0 0.015
            children [
              Box {
                size 0.04 0.165 0.03
              }
            ]
          }
        }
        Camera {
          translation 0 0 0.015
          name "rgb_camera"
          fieldOfView 1.0472
          width 1280
          height 720
        }
        RangeFinder {
          translation 0 0 0.015
          name "depth_camera"
          fieldOfView 1.0472
          width 640
          height 480
          minRange 0.6
          maxRange 8
        }
      ]
    }
    GPS {
      translation -0.365 -0.167 0.302
    }
    Transform {
      translation 0 0 0.39062
    }
    Transform {
      translation 0.2865 -0.20894 0.193
      rotation 0.9238766724085287 -0.3826903371909772 0 3.14159
    }
    Transform {
      translation -0.2865 0.20894 0.193
      rotation 0.3826903371909772 0.9238766724085287 0 3.14159
    }
  ]
  name "Summit_XL_Steel"
  boundingObject Group {
    children [
      Transform {
        translation 0 0 0.3
        children [
          Box {
            size 0.719 0.57861 0.18
          }
        ]
      }
      Transform {
        translation 0 0 0.176
        children [
          Box {
            size 0.524 0.27 0.06495
          }
        ]
      }
      Transform {
        translation 0 0 0.046225
        children [
          Box {
            size 0.69 0.27 0.1945
          }
        ]
      }
    ]
  }
  physics Physics {
    density -1
    mass 125
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      1.391 6.853 6.125
      0.004 0 0
    ]
  }
  controller "lidar_test_controller"
}
